src/core/Akka.Streams.TestKit/TestSubscriber_Shared.cs

```diff
@@ -0,0 +1,375 @@
+﻿// //-----------------------------------------------------------------------
+// // <copyright file="TestSubscriber_Shared.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.TestKit;
+using Reactive.Streams;
+
+namespace Akka.Streams.TestKit
+{
+    public static partial class TestSubscriber
```

Comment:
This is the actual code for all the TestSubscriber methods, it is being called in 3 different places for different flavors, the old backward compatible code, some of the async codes, and in the fluent async builder code